dds/DCPS/DiscoveryBase.h

```diff
@@ -1323,6 +1323,38 @@ namespace OpenDDS {
           reader_local = true;
         }
 
+        // for Xtypes, check consistency
+        XTypes::TypeIdentifier writer_type_id = writer_type_info.minimal.typeid_with_size.type_id;
+        XTypes::TypeIdentifier reader_type_id = reader_type_info.minimal.typeid_with_size.type_id;
+        if (writer_type_id.kind() != XTypes::TK_NONE && reader_type_id.kind() != XTypes::TK_NONE) {
+          //look up topic name
+          TopicNameMap::iterator name_iter = topic_names_.find(writer); //TODO find reader too?
+          if (name_iter != topic_names_.end()) {
+            //find TopicDetails instance in topics_
+            typename OPENDDS_MAP(OPENDDS_STRING, TopicDetails)::iterator TDiter = topics_.find(name_iter->second);
```

Comment:
```suggestion
            typename OPENDDS_MAP(OPENDDS_STRING, TopicDetails)::iterator td_iter = topics_.find(name_iter->second);
```
I think we prefer snake_case or camelCase for names.